libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m

```diff
@@ -82,7 +82,7 @@ + (instancetype) sharedInstanceWithUser:(SFUserAccount *) userAccount {
         }
         NSString *key = SFKeyForUserAndScope(userAccount, SFUserAccountScopeCommunity);
         id analyticsMgr = analyticsManagerList[key];
-        if (!analyticsMgr) {
+        if (!analyticsMgr && key) {
```

Comment:
If key is never nil, what's the hard of checking here? Reason for this asking is if I remember correctly, XCode would warn you if you don't check map key or value when you save it as kind of analysis warning